### PR TITLE
[CDAP-8376] Log Viewer WebApp + LogAction

### PR DESCRIPTION
**Pull request by elfenheart**
JIRA: https://issues.cask.co/browse/CDAP-8376

### CODE REVIEW

File: cdap-ui/app/cdap/components/FastAction/LogAction/index.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        target="_tab"
+        className="btn btn-link"
+      >
+        <span className="fa fa-file-text-o" />
+      </a>
+    );
+
+    return (
+      <span>
+        {this.state.runId ? renderLog : renderDisabled}
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/FastAction/LogAction/index.js]: Missing tooltip?

File: cdap-ui/app/logviewer/main.js [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+        'cask-angular-theme',
+        'cask-angular-focus',
+        'ngCookies'
+      ]).name,
+
+      angular.module(PKG.name+'.filters', [
+        PKG.name+'.services',
+        'cask-angular-capitalize'
+      ]).name,
+
```
Comment (ajainarayanan) [inline cdap-ui/app/logviewer/main.js:44]: We could probably remove half of these. Will drastically reduce the build size though.

File: cdap-ui/Gulpfile.js [hunk 3]
```diff
@@ -273,6 +274,10 @@ gulp.task('watch:js:app:tracker', function() {
   return getExtensionBuildPipeline('tracker')
     .pipe(plug.livereload({ port: 35728 }));
 });
+gulp.task('watch:js:app:logviewer', function() {
+  return getExtensionBuildPipeline('logviewer')
+    .pipe(plug.livereload({ port: 35728 }));
```
Comment (ajainarayanan) [inline cdap-ui/Gulpfile.js:279]: Might want to set this to `35729`? watch might not work for both tracker and logviewer

**Review by ajainarayanan — COMMENTED** [review]
Couple of comments but the log viewer integration looks 🎆 💯

File: cdap-ui/Gulpfile.js (hunk 3 above)
Comment (elfenheart) [reply cdap-ui/Gulpfile.js:279]: everything in here seems to be `35728`, that's why I was not sure..

**Review by ajainarayanan — APPROVED** [review]
Talked to @elfenheart Code 👍 LGTM. We might need to rephrase error message on any unknown entity. If that can come in another PR 💯 Lets merge this one!

**Comment by elfenheart** [comment]
Build: http://builds.cask.co/browse/CDAP-DRC5627-8

**Pull request merged**